### PR TITLE
Fix Issue with prune trial result status 

### PR DESCRIPTION
**Pull request by khansaad**
Signed-off-by: Saad Khan <saakhan@redhat.com>

This PR aims to fix the issue #17 .
Updated the list having the supported types in the utils.py file

### CODE REVIEW

**Comment by khansaad** [comment]
[benchmark.log](https://github.com/kruize/hpo/files/9150502/benchmark.log)
[experiment-output.csv](https://github.com/kruize/hpo/files/9150505/experiment-output.csv)

@kusumachalasani  I have updated the code and ran the benchmark as you suggested.
I have attached the log file here. Please check and confirm if anything else needs to be changed.

**Review by dinogun — CHANGES_REQUESTED** [review]
Looks like we need to update the API.md as well. Also update API.md on what each of these result status means and when should the user be using each of them

File: src/utils.py [hunk 1]
```diff
@@ -18,7 +18,8 @@
 class HPOSupportedTypes:
     DIRECTIONS_SUPPORTED = ("minimize", "maximize")
     VALUE_TYPES_SUPPORTED = ("double", "int", float)
-    ALGOS_SUPPORTED = ("optuna_tpe", "optuna_tpe_multivariate", "optuna_skopt")
+    OPTUNA_ALGOS = ("optuna_tpe", "optuna_tpe_multivariate", "optuna_skopt")
+    ALGOS_SUPPORTED = (OPTUNA_ALGOS)
     TRIAL_RESULT_STATUS = ("success", "failure", "error", "prune")
```
Comment (chandrams) [inline src/utils.py:23]: @khansaad - Shouldn't we be removing prune here and internally pass prune to optuna when result status is failure?

File: src/bayes_optuna/optuna_hpo.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def perform_experiment(self):
             trial_result = self.trialDetails.trial_result
         finally:
             self.resultsAvailableCond.release()
-        trial_result = "prune"
+        trial_result = "failure"
```
Comment (dinogun) [inline src/bayes_optuna/optuna_hpo.py:115]: Wondering, will this not cause us to always return failure?

Comment (khansaad) [reply src/bayes_optuna/optuna_hpo.py]: removed it now as it was added by mistake

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**